primitives/blockchain/src/backend.rs

```diff
@@ -59,19 +59,23 @@ pub trait HeaderBackend<Block: BlockT>: Send + Sync {
 
 	/// Get block header. Returns `UnknownBlock` error if block is not found.
 	fn expect_header(&self, id: BlockId<Block>) -> Result<Block::Header> {
-		self.header(id)?.ok_or_else(|| Error::UnknownBlock(format!("{}", id)))
+		self.header(id)?.ok_or_else(|| Error::UnknownBlock(format!("Expect header: {}", id)))
```

Comment:
Semantics is unclear, maybe provide more explanation in the error message, just leave id here there, or create new error enum variant?

Reply:
Not sure what exactly you expect, `UnknownBlock` error is used pretty commonly everywhere, I wanted to add some extra context to the error message so that we have more info tracing where it comes from. Such `UnknownBlock` errors are constructed in other places in the codebase already (giving some extra description).

Unless you mean to just change `"Expect header" (the function name) to something more like: "Missing header when calling `expect_header`", is that what you mean?

Reply:
Yeah, roughly.
Ok, so
`UnknownBlock(<Specifying which block>)` - is a clear semantics
`UnknownBlock(<Explaing situation that lead to that error>)` - is a clear semantics
`UnknownBlock(<Expect header?/Require header?>)` - not so much clear

 But not strong on this, especially if it is just for tracking error origin/tracing.